coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -107,6 +108,7 @@ func (t Tags) Valid() error {
 }
 
 func NewServer(
+	ctx context.Context,
```

Comment:
I think we could name this differently, maybe `lifecycleCtx`, or something better. Why? Usually a context is only relevant for the execution of a function. Here I would expect `ctx` not to have any effect after `NewServer` returns, but instead the server keeps it around.